### DISCUSSION TITLE
Thinking Through Local Changes

### DISCUSSION THREAD

**Discussion by moo** [dabreegster/abstreet#490] [General]
Hi all! Dustin presented A/B Street at a Greenwood-Phinney Greenways Meeting, and I've been pretty excited about the possibilities. I'm not a dev, but I love visualizing tools - [Seattle Collisions](http://seattlecollisions.timganter.io/collisions) is pretty helpful for showing that aspect. I've basically thrown out tinkering in Cities: Skylines in favor of A/B Street! I'm also working on a few video projects, and think it will be great to cut together live footage on streets with some simulated use case footage. It would be really cool to give some examples, and then let people play around and tweak the settings! 

So, I've been thinking of getting the most current city street improvements up-to-date in OSM, and then thinking about how A/B could be used to show other upcoming changes. These are the most exciting, imho: 
- [Northgate Pedestrian & Bike Bridge](http://www.seattle.gov/transportation/projects-and-programs/programs/bridges-stairs-and-other-structures/bridges/northgate-pedestrian-and-bicycle-bridge) - this is already in OSM as a [footway bridge under construction](https://www.openstreetmap.org/way/842065364) - if this could be added, it could be set to "close for construction" but people could toggle it "open" to try it out! 
- [Climate Pledge Arena](https://www.seattle.gov/transportation/projects-and-programs/programs/bike-program/protected-bike-lanes/1st-ave-and-broad-st-complete-street-extension-project) - maybe a challenge to "flush" a stadium-sized crowd? 
- [12th Ave S](https://www.seattle.gov/transportation/projects-and-programs/safety-first/vision-zero/projects/12th-ave-s)/Jose Rizal Bridge - I've already started [updating this in OSM](https://www.openstreetmap.org/way/428246781), (A/B seems to have put parking on the bridge, but it's getting there!)

I've mentioned adding #488 Traffic Diverters as a possible tool/widget, and I think a Center Turn Lane option could be helpful. 
What are you all interested in?

**Comment by dabreegster** [comment]
Thanks for writing these up!

> Northgate Pedestrian & Bike Bridge

We could certainly do something ahead of #67 to bring in the under-construction bridge, and let people see what happens when it's finished. It looks like the east end of the bridge just ends in empty space right now; are there any published plans showing how it'll connect to the platform and down to the ground?
![Screenshot from 2021-02-01 10-00-28](https://user-images.githubusercontent.com/1664407/106498620-65183000-6474-11eb-8fba-3b30669a014f.png)
Aside from just visualizing people walking and biking over the new bridge, we could measure the expected number of people using the bridge (based on 2014 Soundcast demand models, which of course don't account for all the new development around Northgate, and more people headed that way for the light rail) and visualize some individual trips that can use it as a shortcut instead of riding with traffic on 1st Ave NE or the other alternatives.

> Climate Pledge Arena - maybe a challenge to "flush" a stadium-sized crowd?

I love this as an idea for a challenge mode! We could estimate how many people are headed to and from the stadium on a game day and add that demand to the existing schedule. The player's challenge is to adjust signal timing, maybe close/reverse some lanes, and make other adjustments to both get people in/out of the stadium quickly, while keeping it as easy as possible to cycle there.

> 12th Ave S/Jose Rizal Bridge

Any ideas what to demonstrate or measure here -- just visualizing people moving through the space? This might be one of the easiest things to set up. I haven't imported Seattle OSM data in a few weeks to pick up your changeset, but if we wanted to focus on this one, I can start a dedicated issue to fix up things like parking in the area.

**Reply by moo** [reply]
**Northgate Bridge**
> It looks like the east end of the bridge just ends in empty space right now; are there any published plans showing how it'll connect to the platform and down to the ground?

I added links to the above, and finished the [bridge to the ground on OSM](https://www.openstreetmap.org/way/842065364). I love showing the demand models! 

**Climate Pledge Arena**
I should have noted that I'm interested in both comparing modes and seeing if other road configurations could be more effective. I'll look into challenges - is there a way to use incentives such as "transit fare included with ticket"? 

**12th Ave S/Jose Rizal Bridge**
This is already in-progress, so I just wanted to update it! It'd be interesting to compare traffic, especially for buses going from 4 to 2 general travel lanes, but I honestly don't ever remember it getting really backed up.

**Reply by dabreegster** [reply]
> 12th Ave S/Jose Rizal Bridge

I removed the inferred parking: https://www.openstreetmap.org/changeset/98671294. https://a-b-street.github.io/docs/side_projects/parking_mapper.html has more info on how this gets inferred from the blockface dataset when not in OSM.

So the goal for studying this section could just be to get an idea of how the traffic looks. One big issue here is that A/B Street isn't importing most bus routes right now, due to some long-standing bugs. I'm unlikely to have time to work on that anytime soon. If you think it'd be worth a blog post / something else focusing on this area, could you draw a boundary with geojson.io/ around the area that'd be relevant to show?

**Reply by dabreegster** [reply]
> Northgate Bridge

Thanks for updating the bridge! I'll pull down new OSM data soon and make some changes to import the bridge as the construction lane type by default. Same as the other thread, could you use geojson.io to draw the study area near here to demonstrate in the blog post? I'm only familiar with the sound end of this, and would probably put the boundary around 90th or so. Probably Aurora on the west end, Roosevelt on the east, and something in between Northgate and Haller Lake on the north?

After we get the map here set up, I'll make any interventions needed to get the simulation working without gridlock. Then we can start thinking through all the questions an article could ask.

**Comment by michaelkirk** [comment]
>> Climate Pledge Arena - maybe a challenge to "flush" a stadium-sized crowd?
>
> I love this as an idea for a challenge mode! We could estimate how many people are headed to and from the stadium on a game day and add that demand to the existing schedule. The player's challenge is to adjust signal timing, maybe close/reverse some lanes, and make other adjustments to both get people in/out of the stadium quickly, while keeping it as easy as possible to cycle there.

I also really like this idea!

**Reply by dabreegster** [reply]
Some next steps on the CPA challenge:

- [ ] Draw a study area with geojson.io -- the downtown boundary isn't bad, but probably has too much of cap hill, and maybe we can translate everything north a bit
- [ ] Look around for estimated number of people going to the arena on an event day, and if they have it, where they're coming from or any mode split guesses
- [ ] think through "levels" of the challenge, and how to score them. For example, maybe first you can't influence mode, and you just have to re-time traffic lights to get things going. Then on another level, we define some "convenience cost", where if you make the bus run consistently faster, people will take it. (Specifics here may matter if you have an intended "solution" in mind. I'm not offhand familiar with bus routes nearby. Maybe this challenge actually winds up being a process of getting people onto the monorail from the downtown side.)

> is there a way to use incentives such as "transit fare included with ticket"?

We don't have any modeling whatsoever of cost (like for toll roads or paid parking). And right now, no mode choice change, just what Soundcast prescribes. So maybe this project forces us to make a real start on https://github.com/a-b-street/abstreet/issues/448
